File: stack/stack.cpp
//stack data structure

#include<iostream>

using namespace std;

#define MAX 10

class Stack
{
    int top; //index of the top element
    int a[MAX]; //MAX- maximum size of the stack

    public:

    Stack()
    {
        top = -1;
    }

    void push(int element)
    {
        if (top >= MAX)
            cout<<"Stack Overflow "; //stack is full and hence cannot insert an element
        else
        {
            top++;
            a[top] = element;
            cout<<"Push Operation Successful ";
        }
    }

    int pop()
    {
        if (top == -1)
        {
            cout<<"Stack Underflow "; //No element present in stack to pop
            return -1;
        }
        else
        {
            int popped_element = a[top];
            top--;
            cout<<"Pop operation successful ";
            return popped_element;
        }
    }

    int show_top()
    {
        cout<<"The top element is "<<a[top];
    }

    bool isEmpty()
    {
        if (top == -1)
            return true;
        else
            return false;
    }

    void show_stack()
    {
        cout<<"The content of the stack is as follows: ";
        for(int i = 0; i<=top; i++)
            cout<<a[i]<<"  ";
    }

};

int main()
{
    Stack s;
    int operation;
    char choice = 'y';
    while (choice == 'y')
    {
      cout<<"Enter your choice:\n1.Push\n2.Pop\n3.Show Top\n4.isEmpty\n5.Show Stack\n";
      cin>>operation;

      switch(operation)
      {
          case 1: int element;
                  cout<<"Enter the element to be pushed ";
                  cin>>element;
                  s.push(element);
                  break;

          case 2: s.pop();
                  break;

          case 3: s.show_top();
                  break;

          case 4: s.isEmpty();
                  break;

          case 5: s.show_stack();
                  break;
      }

      cout<<"Press 'y' to continue and 'n' to quit ";
      cin>>choice;
    }

}

File: DataStructuresPython/stack.py
# Implementation of Stack Data Structure using Python3
# In a stack, the top element or the first element is removed
# while new elements are added to the end of the list.

def insert(stack,n):
  if len(stack) == 10:
    print('Overflow: Stack size full. Cannot push element.')
  else:
    stack.append(n)
    print('Element has been pushed.')
  return(stack)

def remove(stack):
  if len(stack) == 0:
    print('Underflow : No element is present that can be popped.')
  else:
    print('The stack has been popped : ', stack.pop(0))
  return(stack)

def showStack(stack):
  print(stack)


def main():
  stack = []
  while(True):
    choice = int(input('Please enter the operation that you would like to perform on the stack : \n1.Push \n2.Pop \n3.Show \n4.Quit : '))
    if choice == 1:
      element = int(input('Enter the element that you wish to push : '))
      print(insert(stack,element))
    elif choice == 2:
      remove(stack)
    elif choice == 3:
      showStack(stack)
    elif choice == 4:
      print('Thank you!')
      break
    else:
      print('Invalid Choice!')


if __name__ == '__main__':
  main()